File: RemoveDuplicatesFromSortedList.py
# Definition for singly-linked list.
from typing import List 
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head 

        # set up a dummy head 

        dummy = ListNode(val=None, next=head)
        pre, cur = dummy, head 
        
        while cur:
            if pre.val == cur.val:
                # if pre's value is equal to cur.value 
                # connect pre's next to cur's next 
                pre.next = cur.next 
                cur = cur.next 
            else:
                # if value is different, move pointer
                pre, cur = cur, cur.next 
            
        return head

        

File: RemoveDuplicatesFromSortedList.go
package main
 // Definition for singly-linked list.

 type ListNode struct {
	Val int
	Next *ListNode
}

func deleteDuplicates(head *ListNode) *ListNode {
	   if head == nil {
		 return nil
	   }

	   dummy := &ListNode {
		 Val: -1,
		 Next: head,
	   }
	   pre := dummy 
	   cur := head 

	   for cur != nil {
			 if pre.Val != cur.Val {
			   pre = cur
			   cur = cur.Next
			 } else {
			   pre.Next = cur.Next
			   cur = cur.Next
			 }
	   }

	   return head
}